### PR TITLE
feat: Create multiple Task From Issue Doctype

### PR DESCRIPTION
**Pull request by viralkansodiya**
Create multiple Task From Issue Doctype(Custom Button)

For check in more details please view this video
https://youtu.be/R6eL8dkOWUg?si=nzs5uWAerQT7oWxL
[](https://youtu.be/R6eL8dkOWUg?si=nzs5uWAerQT7oWxL)

### CODE REVIEW

**Comment by viralkansodiya** [comment]
[](https://youtu.be/R6eL8dkOWUg?si=nzs5uWAerQT7oWxL)

File: erpnext/support/doctype/issue/issue.py [hunk 1]
```diff
@@ -391,3 +398,32 @@ def get_holidays(holiday_list_name):
 	holiday_list = frappe.get_cached_doc("Holiday List", holiday_list_name)
 	holidays = [holiday.holiday_date for holiday in holiday_list.holidays]
 	return holidays
+
+
+@frappe.whitelist()
+def create_new_task_from_issue(self, assign_to):
```
Comment (ankush) [inline erpnext/support/doctype/issue/issue.py:404]:
```suggestion
def create_new_task_from_issue(issue, assign_to):
```

`self` is a convention for objects to refer to self. It shouldn't be used like this.

File: erpnext/support/doctype/issue/issue.py [hunk 2]
```diff
@@ -391,3 +398,32 @@ def get_holidays(holiday_list_name):
...
+	self = frappe.json.loads(self)
```
Comment (ankush) [inline erpnext/support/doctype/issue/issue.py:405]: What is this? `frappe.json` isn't a thing. You're indirectly using json module.

File: erpnext/support/doctype/issue/issue.py [hunk 3]
```diff
@@ -391,3 +398,32 @@ def get_holidays(holiday_list_name):
...
+	doc = frappe.new_doc("Task")
+	doc.subject = self.get("subject")
+	doc.issue = self.get("name")
+	doc.description = self.get("description")
+	doc.priority = self.get("priority")
+	doc.status = self.get("status")
+	doc.project = self.get("project")
```
Comment (ankush) [inline erpnext/support/doctype/issue/issue.py:412]: too much repetition. Use `get_mapped_doc` instead which is used everywhere for mapping one document to another.

File: erpnext/support/doctype/issue/issue.py [hunk 4]
```diff
@@ -391,3 +398,32 @@ def get_holidays(holiday_list_name):
...
+	doc.save()
+	args = {
+		"assign_to": [assign_to],
+		"doctype": "Task",
+		"name": doc.name,
+	}
+	add(args)
+	frappe.msgprint("Task Created Successfully , {0}".format(get_link_to_form("Task", doc.name)))
```
Comment (ankush) [inline erpnext/support/doctype/issue/issue.py:420]: https://frappeframework.com/docs/user/en/translations 
```suggestion
	frappe.msgprint(_("Task Created Successfully: {0}").format(get_link_to_form("Task", doc.name)))
```

File: erpnext/support/doctype/issue/issue.py [hunk 5]
```diff
@@ -391,3 +398,32 @@ def get_holidays(holiday_list_name):
...
+
+
+@frappe.whitelist()
+def get_if_task_exist(self):
+	self = frappe.json.loads(self)
+	if name := frappe.db.exists("Task", {"issue": self.get("name")}):
+		return name
+	else:
+		return
```
Comment (ankush) [inline erpnext/support/doctype/issue/issue.py:429]:
```suggestion
def get_if_task_exist(issue):
	return bool(frappe.db.exists("Task", {"issue": issue}))
```

Why do you need this much code? This method can be removed entirely. There's `frappe.db.exist` on client side too.

File: erpnext/support/doctype/issue/issue.js [hunk 6]
```diff
@@ -145,4 +146,70 @@ frappe.ui.form.on("Issue", {
...
+                    args: {
+                        self:frm.doc,
+                        assign_to:data.assign_to
+                    },
+                    callback: function(r) {
+                        if(!r.exc) {
+
+                            d.hide();
+                        }
+                        d.hide();
```
Comment (ankush) [inline erpnext/support/doctype/issue/issue.js:207]: Why is this required?

File: erpnext/support/doctype/issue/issue.js [hunk 7]
```diff
@@ -145,4 +146,70 @@ frappe.ui.form.on("Issue", {
...
+                        frappe.confirm('Task is Already Created.<br>Do you want to create another Task?',
+                    () => {
+                        frm.trigger('opendiolog')
+                    }, () => {
+
+                    })
+                    }
+                    else{
+                        frm.trigger('opendiolog')
+                    }
```
Comment (ankush) [inline erpnext/support/doctype/issue/issue.js:172]: Format this correctly please, it's not readable at all.

File: erpnext/support/doctype/issue/issue.js [hunk 8]
```diff
@@ -145,4 +146,70 @@ frappe.ui.form.on("Issue", {
 		// 	frm.timeline.wrapper.data("help-article-event-attached", true);
 		// }
 	},
+	refresh:function(frm){
+		frm.trigger('create_button')
+	},
+	create_button:function(frm){
+        if(!frm.doc.__islocal) {
```
Comment (ankush) [inline erpnext/support/doctype/issue/issue.js:153]:
```suggestion
        if(!frm.is_new()) {
```

File: erpnext/support/doctype/issue/issue.py [hunk 9]
```diff
@@ -8,12 +8,19 @@
 import frappe
 from frappe import _
 from frappe.core.utils import get_parent_doc
+from frappe.desk.form.assign_to import add
```
Comment (ankush) [inline erpnext/support/doctype/issue/issue.py:11]:
```suggestion
from frappe.desk.form.assign_to import add as add_assignment
```

Better readability

**Review by ankush — CHANGES_REQUESTED** [review]

**Comment by deepeshgarg007** [comment]
@viralkansodiya Doesn't look like a very high-in-demand feature, maybe keep this as a Client Script/Server Script on the site where this is needed?

**Comment by ankush** [comment]
Also, implicitly creating tasks isn't of much use IMO. It should at least let me say what task I'm supposed to do.

**Comment by viralkansodiya** [comment]
@deepeshgarg007 
In Services companies, clients create an 'Issue'.  Customer Support team will review and provide information and close the issue when possible. When the issue raised by client requires some additional work, a Task is created and this task maybe used even in timesheets. 

So this feature is very useful for services companies. 

Request you to kindly reconsider merging this PR once I implement changes suggested by Ankush.

**Pull request closed without merging**